lib/fake_ifdh.py

```diff
@@ -131,7 +133,13 @@ def getToken(role: str = DEFAULT_ROLE, verbose: int = 0) -> str:
         tokenfile = f"{tmp}/bt_token_{issuer}_{role}_{pid}"
         os.environ["BEARER_TOKEN_FILE"] = tokenfile
 
-    if not checkToken(tokenfile):
+    try:
+        token_ok = checkToken(tokenfile)
+    except:
+        # if anything goes wrong checking, its not okay, get a fresh one
+        token_ok = False
+
+    if not token_ok:
```

Comment:
I like this a lot better - thanks Marc!  

One last nitpick (but definitely not a blocker for approval IMO):

Given this new logic, what would you think about, on line 154, changing the last `checkToken` check from this:

```python
if checkToken(tokenfile):
    return tokenfile
raise PermissionError("blahblah")
```
to:

```python
if not checkToken(tokenfile):
    raise PermissionError("blahblah")
```

This way:

1. If that second `checkToken` fails, we get the `PermissionError` raised
2. If an `Exception` is raised from `checkToken`, it gets raised anyway, and 
3. If nothing goes wrong, this clause is skipped, and the `return tokenfile` on what is now line 157 gets executed.

I think that small change might make the logic a little clearer given the context starting from line 151, so that we're only returning in that clause if there's an issue.  In context, that'd look like:

```python
    if not token_ok:
        do_the_htgettoken_and_role_stuff()
        res = os.system(cmd)
        if res != 0:
            raise PermissionError(f"Failed attempting '{cmd}'")
        if not checkToken(tokenfile):
            raise PermissionError(f"Failed validating token from '{cmd}'")
    return tokenfile
```

Reply:
Thanks!